internal/engine/execution/schema_v07.go

```diff
@@ -0,0 +1,224 @@
+package execution
+
+import (
+	"encoding/json"
+
+	"github.com/kwilteam/kwil-db/core/types"
+)
+
+// this file contains the schema from Kwil v0.7, which is used for proper migration to
+// the new schema in v0.8
+
+// convertV07Schema converts a v0.7 schema to a v0.8 schema
+func convertV07Schema(bts []byte) (*types.Schema, error) {
```

Comment:
This function and this file look fairly straightforward.  Some suggestions though: https://github.com/jchappelow/kwil-db/commit/15607b09c86bb3b897098d6f4df28cef7e0e8548
The only thing that looks like a bug is the "bool" vs. "BOOLEAN" change in the data type string.  Also case for all the types, but the code uses `strings.ToLower` a lot so case probably wouldn't matter.